src/main/services/csfloat/csfloatSocket.ts

```diff
@@ -0,0 +1,515 @@
+      if (trade.state !== EStatusTradeCSFLOAT.PENDING) continue;
+
+      if (!trade.wait_for_cancel_ping) continue;
+
+      const tradeOffer = allTradeOffers.find(
+        (item) => item.id === trade.steam_offer.id
+      );
+
+      if (
+        tradeOffer &&
+        (tradeOffer.state === ETradeOfferStateCSFloat.Active ||
+          tradeOffer.state === ETradeOfferStateCSFloat.Accepted)
+      )
+        continue;
+
+      try {
+        await this._csFloatClient.pingCancelTrade({
+          trade_id: trade.id,
+          steam_id: this.steamIDBase64,
+        });
+      } catch (error) {
+        console.error(
+          `failed to send cancel ping for trade ${trade.id}`,
+          error
+        );
+      }
+    }
+  }
+
+  private async autoAcceptTrades(
+    tradesInQueue: ITradeFloat[],
+    tradeOffers: IGetTradeOffersResponde
+  ): Promise<void> {
+    if (tradesInQueue.length < 1) return;
+
+    const tradeOffersSent = tradeOffers.sent;
+    const itemsTradables: CEconItem[] = await new Promise((resolve, reject) => {
+      this.emit("getInventory", (items: CEconItem[], error) => {
+        if (error) reject(error);
+        resolve(items);
+      });
+    });
+
+    if (!itemsTradables) return;
+
+    for (const tradeInQueue of tradesInQueue) {
+      const hasMatchingItem = itemsTradables.some(
+        (item) => item.assetid === tradeInQueue.contract.item.asset_id
+      );
+
+      if (!hasMatchingItem) continue;
+      let alreadyHasOffer: boolean;
+
+      for (const tradeOffer of tradeOffersSent) {
+        for (const item of tradeOffer.itemsToGive) {
+          if (item.assetid === tradeInQueue.contract.item.asset_id) alreadyHasOffer = true
+        }
+      }
+
+      if (!alreadyHasOffer) {
+        await this._csFloatClient
+          .acceptTradesInFloat(tradeInQueue.id)
+          .then(() => {
+            this.emit("notifyWindows", {
+              title: `CSFLOAT - Item: ${tradeInQueue.contract.item.item_name}`,
+              body: `Accepted item!`,
+            });
+          })
+          .catch(() => {
+            this.emit("notifyWindows", {
+              title: `CSFLOAT!`,
+              body: `Item: ${tradeInQueue.contract.item.item_name}, Cannot be accepted!`,
+            });
+          });
+      }
+    }
+  }
+
+
+  private async sendOffer(
+    tradesInPending: ITradeFloat[],
+    tradeOffers: IGetTradeOffersResponde
+  ): Promise<void> {
+    if (tradesInPending.length < 1) return;
+    const tradeOffersAlreadySent = tradeOffers.sent;
+
+    const tradesActives = tradeOffersAlreadySent.filter(
+      (trade) => trade.state === ETradeOfferStateCSFloat.Active
+    );
+
+    for (const tradePending of tradesInPending) {
+
+      const itemToGive = tradePending.contract.item.asset_id.toString();
+      const alreadySentItems = tradesActives.flatMap((trade) =>
+        trade.itemsToGive.map((item) => item.assetid.toString())
+      );
+
+      if (alreadySentItems.includes(itemToGive)) return;
```

Comment:
This early return stop the function even if there are more trades to send
```suggestion
      if (alreadySentItems.includes(itemToGive)) continue;
```